cardano/src/block/verify_chain.rs

```diff
@@ -87,19 +105,33 @@ impl ChainState {
             Block::BoundaryBlock(_) => {}
 
             Block::MainBlock(blk) => {
-                let slot_id = blk.header.consensus.slot_id.slotid as usize;
+                let epoch_id = blk.header.consensus.slot_id.epoch;
+                let epoch_with_ebb = match &self.last_boundary_block_epoch {
+                    Some(last_ebb_epoch) => epoch_id == *last_ebb_epoch,
+                    _ => false,
+                };
 
-                if slot_id >= self.slot_leaders.len() {
-                    return Err(Error::NonExistentSlot);
-                }
+                if epoch_with_ebb {
+                    // If epoch contains EBB - validate block using the leader-list
+
+                    let slot_id = blk.header.consensus.slot_id.slotid as usize;
+
+                    if slot_id >= self.slot_leaders.len() {
+                        return Err(Error::NonExistentSlot);
+                    }
+
+                    let slot_leader = &self.slot_leaders[slot_id];
 
-                let slot_leader = &self.slot_leaders[slot_id];
+                    // Note: the block signature was already checked in
+                    // verify_block, so here we only check the leader key
+                    // against the genesis block.
+                    if slot_leader != &address::StakeholderId::new(&blk.header.consensus.leader_key)
+                    {
+                        return Err(Error::WrongSlotLeader);
+                    }
+                } else {
 
-                // Note: the block signature was already checked in
-                // verify_block, so here we only check the leader key
-                // against the genesis block.
-                if slot_leader != &address::StakeholderId::new(&blk.header.consensus.leader_key) {
-                    return Err(Error::WrongSlotLeader);
+                    // TODO: validate OBFT leader
```

Comment:
No OBFT leader validation present. We don't have the documentation yet on how it's done.